quattor/functions/validation.pan

```diff
@@ -157,3 +156,38 @@ function valid_absolute_file_paths = {
     };
     true;
 };
+
+@documentation{
+The Infiniband hardware address is a series of twenty bytes encoded as hex values
+and separated with a colon or a hyphen.  Within a value you must
+use a consistent separator.
+
+Ex.: 80-06-00-48-FE-80-00-00-00-00-00-00-00-00-00-00
+    80:06:00:48:FE:80:00:00:00:00:00:00:00:00:00:00
+
+Both upper and lower-case hex digits are accepted.
+}
+function is_ibhwaddr = {
+    # Check cardinality and type of argument.
+    if (ARGC != 1 || !is_string(ARGV[0]))
+        error("usage: is_ibguid(string)");
+
+    return(match(ARGV[0],
+        '^[\dA-Fa-f]{2}([:-])[\dA-Fa-f]{2}(\1[\dA-Fa-f]{2}){18}$'));
+};
+
+@documentation{
+The Infiniband guid is a series of 16 chars preceded by 0x.
+
+Ex.: 0x0002c9030002fb06
+
+Both upper and lower-case hex digits are accepted.
+}
+function is_ibguid = {
```

Comment:
if you only need a type, make a type. in panc 10.4, you'll be able to make test functions from any type if you'd ever need one.